### DISCUSSION TITLE
Optional: Using DO Databases instead of Droplet volumes

### DISCUSSION THREAD

**Discussion by RicoFio** [digitalocean/supabase-on-do#37] [Ideas]
Hi DO curators,

While this might not fall into the main focus of this Supabase DO template, it might be worth adding documentation on how people can setup their Supabase on DO while using the dedicated DO Databases product rather than relying on volumes on Droplets. This is also something that is encouraged by the official Supabase docs [here](https://supabase.com/docs/guides/self-hosting/docker#using-an-external-database).

**Comment by timothymamo** [comment]
Hi @RicoFio, thanks for the suggestion. I have looked into this and agree using the managed DO Database would be a lot better as you wouldn't need to manage updates of the database yourself.

Unfortunately the issue lies within this statement (bold text emphases by myself):
>Note that you need **superuser permission** on the postgres role to perform the initial schema migration. Once completed, the postgres role will be demoted to non-superuser to prevent abuse.

Currently there is no automated way of getting superuser access on a managed DO database. This can be requested via support, I believe, but is obviously a manual process.

This is why we attach a volume to the Droplet, to ensure we have persistent data even when/if the Droplet is restarted (the `postgres` container uses this as its [volume](https://github.com/digitalocean/supabase-on-do/blob/main/packer/supabase/docker-compose.yml#L285) for data storage).

As I wanted to have to make the process as automated as possible, and hence simple, I went for a full `docker-compose` setup, rather than using a DO managed database.

But I'll set time to add documentation to explain the above should people wish to go through the manual process of attaching the DO managed database.

**Reply by RicoFio** [reply]
Hi @timothymamo,
Awesome, that's good to know - in fact, I wasn't aware of that fine-print. It makes sense to use a volume instead then.
Thanks a lot for taking your time to follow up on this though!
